scripts/ci/jenkins_diff.sh

```diff
@@ -0,0 +1,184 @@
+  terraform_workspace=${3:-$terraform_workspace}
+  usage_file=${4:-$usage_file}
+  config_file=${5:-$config_file}
+  fail_condition=${7:-$fail_condition}
+
+  # Validate fail_condition
+  if ! echo "$fail_condition" | jq empty; then
+    echo "Error: fail_condition contains invalid JSON"
+  fi
+
+  # Set defaults
+  if [ ! -z "$fail_condition" ] && [ "$(echo "$fail_condition" | jq '.percentage_threshold')" != "null" ]; then
+    fail_percentage_threshold=$(echo "$fail_condition" | jq -r '.percentage_threshold')
+  fi
+  fail_percentage_threshold=${fail_percentage_threshold:--1}
+  INFRACOST_BINARY=${INFRACOST_BINARY:-infracost}
+
+  # Export as it's used by infracost, not this script
+  export INFRACOST_LOG_LEVEL=${INFRACOST_LOG_LEVEL:-info}
+  export INFRACOST_CI_DIFF=true
+
+  if [ ! -z "$GIT_SSH_KEY" ]; then
+    echo "Setting up private Git SSH key so terraform can access your private modules."
+    mkdir -p .ssh
+    echo "${GIT_SSH_KEY}" > .ssh/git_ssh_key
+    chmod 600 .ssh/git_ssh_key
+    export GIT_SSH_COMMAND="ssh -i $(pwd)/.ssh/git_ssh_key -o 'StrictHostKeyChecking=no'"
+  fi
+
+}
+
+build_breakdown_cmd () {
+  breakdown_cmd="${INFRACOST_BINARY} breakdown --no-color --format json"
+
+  if [ ! -z "$iac_path" ]; then
+    breakdown_cmd="$breakdown_cmd --path $iac_path"
+  fi
+  if [ ! -z "$terraform_plan_flags" ]; then
+    breakdown_cmd="$breakdown_cmd --terraform-plan-flags \"$terraform_plan_flags\""
+  fi
+  if [ ! -z "$usage_file" ]; then
+    breakdown_cmd="$breakdown_cmd --usage-file $usage_file"
+  fi
+  if [ ! -z "$config_file" ]; then
+    breakdown_cmd="$breakdown_cmd --config-file $config_file"
+  fi
+  echo "$breakdown_cmd"
+}
+
+build_output_cmd () {
+  breakdown_path=$1
+  output_cmd="${INFRACOST_BINARY} output --no-color --format diff --path $1"
+  echo "${output_cmd}"
+}
+
+format_cost () {
+  cost=$1
+    
+  if [ -z "$cost" ] | [ "${cost}" == "null" ]; then
+    echo "-"
+  elif [ $(echo "$cost < 100" | bc -l) = 1 ]; then
+    printf "$%0.2f" $cost
+  else
+    printf "$%0.0f" $cost
+  fi
+}
+
+build_msg () {
+  change_word="increase"
+  change_sym="+"
+  if [ $(echo "$total_monthly_cost < ${past_total_monthly_cost}" | bc -l) = 1 ]; then
+    change_word="decrease"
+    change_sym=""
+  fi
+
+  percent_display=""
+  if [ ! -z "$percent" ]; then
+    percent_display="$(printf "%.0f" $percent)"
+    percent_display=" (${change_sym}${percent_display}%%)"
+  fi
+
+  msg="${msg}\n##### Infracost estimate #####"
+  msg="${msg}\n\n"
+  msg="${msg}Monthly cost will ${change_word} by $(format_cost $diff_cost)$percent_display\n"
+  msg="${msg}\n"
+  msg="${msg}Previous monthly cost: $(format_cost $past_total_monthly_cost)\n"
+  msg="${msg}New monthly cost: $(format_cost $total_monthly_cost)\n"
+  msg="${msg}\n"
+  msg="${msg}Infracost output:\n"
+  msg="${msg}\n"
+  msg="${msg}$(echo "    ${diff_output//$'\n'/\\n    }" | sed "s/%/%%/g")\n"
+  printf "$msg"
+}
+
+build_msg_html () {
```

Comment:
Maybe add a comment so we remember to switch this once https://github.com/infracost/infracost/issues/509 is released

Reply:
Added a todo comment